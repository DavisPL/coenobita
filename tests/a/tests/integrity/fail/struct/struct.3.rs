#![allow(unused_variables)]
#![allow(unused_assignments)]

struct BarBar<'a> {
    number: i32,
    
    #[cnbt::integrity({root}{root,c})]
    string: &'a str
}

fn main() {
    #[cnbt::integrity({root}{root,c})]
    let string = if c::boolean() {
        "something"
    } else {
        "else"
    };

    #[cnbt::integrity({root}{root} struct { number: {*}{*}, string: {root}{root,c} })]
    let b = BarBar {
        number: b::value(),
        string
    };

    #[cnbt::integrity({root}{root})]
    let y = b.string;
}