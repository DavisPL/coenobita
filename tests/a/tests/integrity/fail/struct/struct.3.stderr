error: expected {root}{root}, found {root}{c,root}
  --> tests/integrity/fail/struct/struct.3.rs:26:13
   |
26 |     let y = b.string;
   |             ^^^^^^^^
